src/update.rs

```diff
@@ -0,0 +1,599 @@
+
+    #[test]
+    fn channel_update() {
+        let x_alice = OwnershipKeyPair::new_random();
+        let x_bob = OwnershipKeyPair::new_random();
+
+        let r_alice = RevocationKeyPair::new_random();
+        let r_bob = RevocationKeyPair::new_random();
+
+        let y_alice = PublishingKeyPair::new_random();
+        let y_bob = PublishingKeyPair::new_random();
+
+        let current_balance = Balance {
+            alice: Amount::from_btc(1.0).unwrap(),
+            bob: Amount::from_btc(1.0).unwrap(),
+        };
+
+        let TX_f = FundingTransaction::new(
+            (x_alice.public(), (TxIn::default(), current_balance.alice)),
+            (x_bob.public(), (TxIn::default(), current_balance.bob)),
+        )
+        .unwrap();
+
+        let time_lock = 60 * 60;
+        let TX_c = CommitTransaction::new(
+            &TX_f,
+            (x_alice.public(), r_alice.public(), y_alice.public()),
+            (x_bob.public(), r_bob.public(), y_bob.public()),
+            time_lock,
+        )
+        .unwrap();
+
+        let TX_s = SplitTransaction::new(
+            &TX_c,
+            ChannelBalance {
+                a: (current_balance.alice, x_alice.public()),
+                b: (current_balance.bob, x_bob.public()),
+            },
+        );
+
+        let alice_encsig = TX_c.encsign_once(x_alice.clone(), y_bob.public());
+        let bob_encsig = TX_c.encsign_once(x_bob.clone(), y_alice.public());
+
+        let alice0 = {
+            let current_state = ChannelState {
+                TX_c: TX_c.clone(),
+                encsig_TX_c_self: alice_encsig.clone(),
+                encsig_TX_c_other: bob_encsig.clone(),
+                r_self: r_alice.clone(),
+                R_other: r_bob.public(),
+                y_self: y_alice,
+                TX_s: TX_s.clone(),
+            };
+
+            Party0 {
+                x_self: x_alice.clone(),
+                X_other: x_bob.public(),
+                TX_f: TX_f.clone(),
+                current_state,
+                revoked_states: vec![],
+            }
+        };
+
+        let bob0 = {
+            let current_state = ChannelState {
+                TX_c,
+                encsig_TX_c_self: bob_encsig,
+                encsig_TX_c_other: alice_encsig,
+                r_self: r_bob,
+                R_other: r_alice.public(),
+                y_self: y_bob,
+                TX_s,
+            };
+
+            Party0 {
+                x_self: x_bob,
+                X_other: x_alice.public(),
+                TX_f,
+                current_state,
+                revoked_states: vec![],
+            }
+        };
+
+        let channel_update = ChannelUpdate::Pay(Amount::from_btc(0.5).unwrap());
+        let time_lock = 60 * 60;
+
+        let (alice1, message0) = alice0
+            .propose_channel_update(channel_update, time_lock)
+            .unwrap();
+
+        let (bob1, message1) = bob0.receive_channel_update(message0).unwrap();
```

Comment:
I think the naming of `bob0-bobX` does not add any value to the states anymore. Can we come up with a better naming scheme which reflects what state the party is in? 
E.g. 
`bob_pending_channel_update`
`